python/drned_xmnr/op/config_op.py

```diff
@@ -225,8 +227,19 @@ def run_xslt(self, nso_xml_file, xml_file):
         with open(nso_xml_file, "w+") as outfile:
             nso_xml.write(outfile)
 
-    def xml_to_state(self, trans, xml_file, state_file):
-        trans.load_config(_ncs.maapi.CONFIG_XML, xml_file)
+    def create_state(self, source_file, state_file, flags):
+        try:
+            self.run_with_trans(lambda trans: self.run_create_state(trans, source_file, state_file,
+                                                                    flags), write=True,
+                                no_commit=True)
+        except Exception:
+            raise ActionError(os.path.basename(source_file) + " " +
+                              repr(traceback.format_exception(*sys.exc_info()))
+                              .split("Error:", 1)[1].replace("\\n", "").replace("']", ""))
+        sys.exc_clear()
```

Comment:
What is this idiom supposed to do, that shouldn't be needed, or is it?

Reply:
The purpose here is to parse the MAAPI error message to give the user immediate feedback on why the state failed to load or validate running the check_states action. Using this feedback the user can get an understanding why the check failed to, for example, quickly correct minor issues in the config used to record / load  the state and replace the failing state.

Reply:
Sorry, I should have been more specific (didn't notice that my remark goes to the whole chunk). I was talking about `sys.exc_clear()` call - I noticed you are using that but 1) AFAIK it should not be needed in normal situations 2) it is not portable to Python3.

Reply:
True, leftover from when I was testing the error messages. I Added a "Remove superfluous call to sys.exc_clear()" commit where I removed the sys.exc_clear() calls